pintc/src/main.rs

```diff
@@ -103,6 +103,49 @@ fn main() -> anyhow::Result<()> {
                 }
             };
 
+            // Compute and print the content addresses for the contract and its predicates
+            let compiled_predicate_contract_addresses: Vec<_> = compiled_contract
+                .predicates
+                .iter()
+                .map(essential_hash::content_addr)
+                .collect();
+            let ca = essential_hash::contract_addr::from_predicate_addrs(
+                compiled_predicate_contract_addresses.iter().cloned(),
+                &args.salt.unwrap_or_default(),
+            );
+
+            let pinned_name = filepath
+                .file_stem()
+                .expect("Failed to get file stem")
+                .to_str()
+                .expect("Invalid unicode in contract name");
+            let name_col_w = {
+                let mut name_w = 0;
+                for name in &compiled_contract.names {
+                    let w = name.chars().count();
+                    name_w = std::cmp::max(name_w, w);
+                }
+                pinned_name.chars().count() + name_w
+            };
+            let bold = Style::new().bold();
+            println!(
+                "    {}contract{} {:<name_col_w$} {}",
+                bold.render(),
+                bold.render_reset(),
+                pinned_name,
+                ca,
+            );
+
+            let mut iter = compiled_predicate_contract_addresses
+                .iter()
+                .zip(&compiled_contract.names)
+                .peekable();
+            while let Some((ca, name)) = iter.next() {
+                let name = format!("{}{}", pinned_name, name);
+                let pipe = iter.peek().map(|_| "├──").unwrap_or("└──");
+                println!("         {pipe} {:<name_col_w$} {}", name, ca);
+            }
```

Comment:
I added quite a bit to `main.rs` to print the addresses properly. I do think it's pretty straightforward/clean, so maybe it's not a big deal? 

Let me know if you'd like me to move it elsewhere. I'd probably opt to create a struct with the sole purpose of using the `display` trait to keep everything organized.

Reply:
IMO, `main()` is already a bit big.  It's probably possible to split it up into different functions, though doing that for the sake of it is subjective.  I.e., is it actually clearer when split up?  Is it clearer with every stage after each other?

Using `Display` isn't really necessary though, IMO.  Especially if it's pretty much only used once.

Reply:
Yeah I guess it's fine for now.. you can open an issue if you want to refactor main into smaller pieces.. but it's low priority.

Reply:
I'll hold off on opening an issue. Seems like it'd just be bikeshedding at best.